File: 3653-xor-after-range-multiplication-queries-i/3653-xor-after-range-multiplication-queries-i.java
class Solution {
    int mod= (int)1e9+7;
    public int xorAfterQueries(int[] nums, int[][] queries) {
        for(int i=0; i<queries.length; i++){
            int l= queries[i][0], r= queries[i][1];
            while(l <= r){
                nums[l]= (int)(((long)nums[l] * queries[i][3])%mod);
                l+= queries[i][2];
            }
        }
        int n= nums.length;
        int ans=0;
        for(int i=0; i<n; i++){
            ans^= nums[i];
        }
        return ans;
    }
}

File: 3653-xor-after-range-multiplication-queries-i/3653-xor-after-range-multiplication-queries-i.cpp
class Solution {
public:
    const int mod= 1e9+7;
    int xorAfterQueries(vector<int>& nums, vector<vector<int>>& queries) {
        for(vector<int>& i: queries){
            int l= i[0];
            while(l <= i[1]){
                nums[l]= ((long long)nums[l] * i[3]) % mod;
                l+= i[2];
            }
        }
        int ans=0;
        for(int i: nums) ans^= i;
        return ans;
    }
};